File: app/src/main/java/org/skyfaced/mvp/di/RepositoryModule.java
package org.skyfaced.mvp.di;

import org.skyfaced.mvp.service.network.WaifuRepository;
import org.skyfaced.mvp.service.network.WaifuRepositoryImpl;
import org.skyfaced.mvp.service.network.WaifuService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;

@Module
@InstallIn(SingletonComponent.class)
public final class RepositoryModule {
    @Provides
    @Singleton
    WaifuRepository waifuRepository(WaifuService waifuService) {
        return new WaifuRepositoryImpl(waifuService);
    }
}

File: app/src/main/java/org/skyfaced/mvp/di/RepositoryModule.kt
package org.skyfaced.mvp.di

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import org.skyfaced.mvp.service.network.WaifuRepository
import org.skyfaced.mvp.service.network.WaifuRepositoryImpl
import org.skyfaced.mvp.service.network.WaifuService
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object RepositoryModule {
    @Provides
    @Singleton
    fun waifuRepository(
        waifuService: WaifuService,
    ): WaifuRepository = WaifuRepositoryImpl(waifuService)
}